### PR TITLE
Content for Incremental updates in Drupal

### PR DESCRIPTION
**Pull request by Kalaiselvan88**
My first ADR on incremental updates to Drupal

### CODE REVIEW

File: content/adr/drupal-incremental-updates.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+---
+title: "Incremental updates to Drupal"
+date: "2024-02-29"
```
Comment (hussainweb) [inline content/adr/drupal-incremental-updates.md:3]: Why this date?

File: content/adr/drupal-incremental-updates.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+status: "accepted"
+categories:
+  - Drupal
+---
+
+# Drupal 9
+
+## Context
+
+Currently, we focus on doing major security updates to contributed modules and core as part of maintenance. However, this becomes a challenge when the end of life (EOL) arrives for our current Drupal major version as it results in one large estimation for fixing deprecated custom code, replacing contributed modules, and finally upgrading the core. This causes a huge one-time cost for the clients making it challenging to get budget approvals It also is not in the spirit of smaller incremental development and hence impacts the stability of the application resulting in more efforts in each phase including QA and UAT.
```
Comment (hussainweb) [inline content/adr/drupal-incremental-updates.md:14]: I have only made minor fixes but I think we need to rewrite this. It's confusing at the moment.

```suggestion
Currently, we focus on only doing major security updates to contributed modules and core as part of maintenance. However, this becomes a challenge when the end of life (EOL) arrives for our current Drupal major version as it results in one large estimation for fixing deprecated custom code, replacing contributed modules, and finally upgrading the core. This causes a huge one-time cost for the clients making it challenging to get budget approvals. It is also not in the spirit of smaller incremental development and hence impacts the stability of the application resulting in more efforts in each phase including QA and UAT.
```

File: content/adr/drupal-incremental-updates.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
+---
+title: "Incremental updates to Drupal"
```
Comment (hussainweb) [inline content/adr/drupal-incremental-updates.md:2]: Let's think of a clearer title. When I read this at first, I had no clue what you meant or why this was an ADR.

**Review by hussainweb — CHANGES_REQUESTED** [review]

File: content/adr/drupal-incremental-updates.md (hunk 1 above)
Comment (Kalaiselvan88) [reply content/adr/drupal-incremental-updates.md:3]: Added this based on [when we discussed about this](https://axelerant.slack.com/archives/C06LU6DDVEX/p1709182475884979?thread_ts=1709039712.881909&cid=C06LU6DDVEX) when doing the CDM packages

**Comment by Kalaiselvan88** [comment]
@hussainweb also any reason ADR folder is outside docs? I was thinking if we should add it inside and show as part of menu?

File: content/adr/drupal-incremental-updates.md (hunk 3 above)
Comment (hussainweb) [reply content/adr/drupal-incremental-updates.md]: @Kalaiselvan88, I'm afraid it's still unclear to me if I were to just read the title. I want to convey that this is in context for preparation for major updates and not leaving it until the last minute.

Something to the effect of "Continuous handling of module updates and deprecated API".

Comment (Kalaiselvan88) [reply content/adr/drupal-incremental-updates.md]: @hussainweb check e72861bcad573c71b3932309a8cb7e30cd6a7acb

File: content/adr/drupal-incremental-updates.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
+---
+title: "A Continuous Approach to Module Updates and Deprecated API Handling in Drupal"
+date: "2024-02-29"
+decision: "We will perform incremental updates to fix deprecated code in our custom modules, replacing deprecated contributed modules with latest versions or an alternative rather than one shot upgrade from one Drupal major version to another"
+status: "accepted"
+categories:
+  - Drupal
+---
+
+# Drupal 9
```
Comment (hussainweb) [inline content/adr/drupal-incremental-updates.md:10]:
```suggestion
# A Continuous Approach to Module Updates and Deprecated API Handling in Drupal
```

File: content/adr/drupal-incremental-updates.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
+---
+title: "A Continuous Approach to Module Updates and Deprecated API Handling in Drupal"
```
Comment (hussainweb) [inline content/adr/drupal-incremental-updates.md:2]:
```suggestion
title: "Regularly Update Modules and Remove Deprecated APIs"
```

File: content/adr/drupal-incremental-updates.md [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+date: "2024-02-29"
+decision: "We will perform incremental updates to fix deprecated code in our custom modules, replacing deprecated contributed modules with latest versions or an alternative rather than one shot upgrade from one Drupal major version to another"
+status: "accepted"
+categories:
+  - Drupal
+---
+
+# A Continuous Approach to Module Updates and Deprecated API Handling in Drupal
```
Comment (hussainweb) [inline content/adr/drupal-incremental-updates.md:10]:
```suggestion
# Regularly Update Modules and Remove Deprecated APIs
```

**Review by hussainweb — COMMENTED** [review]
@Kalaiselvan88, please check alternate versions. I am thinking beginning the title with a verb makes more sense when reading the page. See https://deploy-preview-224--axelerant-engg-handbook.netlify.app/adr/.

File: content/adr/drupal-incremental-updates.md (hunk 3 above)
Comment (Kalaiselvan88) [reply content/adr/drupal-incremental-updates.md]: Accepted suggestions

**Review by hussainweb — APPROVED** [review]
Since this is already accepted by the team and in practice, I will merge this. Regularly, ADR's need to go through review.

**Pull request merged**